### ISSUE TITLE
Serial end in Software serial library

### ISSUE THREAD

**Issue by vignesh13404387** [arduino/ArduinoCore-avr#311]
Does Software serial buffer will clear while using SoftwareSerial.end()?  What happens when we create a software serial object and when will the buffer size allocate?

**Comment by swap9820** [comment]
Yes, Serial.end() does clear out the serial buffer but is not recommended as it may fragment the memory. Buffer Size is allocated as soon as the existing buffer is flushed,
